adapter/src/hermes/adapter/stdio/metadata_manager.h

```diff
@@ -0,0 +1,115 @@
+#ifndef HERMES_ADAPTER_METADATA_MANAGER_H
+#define HERMES_ADAPTER_METADATA_MANAGER_H
+
+/**
+ * Standard headers
+ */
+#include <ftw.h>
+
+#include <cstdio>
+#include <unordered_map>
+
+/**
+ * Internal headers
+ */
+#include <hermes/adapter/stdio/common/constants.h>
+#include <hermes/adapter/stdio/common/datastructures.h>
+
+namespace hermes::adapter::stdio {
+/**
+ * Metadata manager for STDIO adapter
+ */
+class MetadataManager {
+ private:
+  /**
+   * Private members
+   */
+  /**
+   * Maintain a local metadata FileID structure mapped to Adapter Stats.
+   */
+  std::unordered_map<FileID, AdapterStat> metadata;
+  /**
+   * hermes attribute to initialize Hermes
+   */
+  std::shared_ptr<hapi::Hermes> hermes;
+  /**
+   * references of how many times hermes was tried to initialize.
+   */
+  size_t ref;
```

Comment:
Shouldn't this be a `std::atomic<size_t>`?

Reply:
Done